### DISCUSSION TITLE
Workflow with powershell and wsl

### DISCUSSION THREAD

**Discussion by pwsandoval** [wezterm/wezterm#3271] [Q&A]
It's an excellent terminal and I'm almost done customizing it.

But I'm not sure how to achieve a workflow similar to the one I had when I used Windows Terminal. This discussion https://github.com/wez/wezterm/discussions/2093 helped me understand a few things.

Although my flow is a little different. I use WSL (ubuntu) and Powershell (7.3.3)

**With WSL I have:**

| Status                      | Action                                                       | Want = Happens                                         |
|----------------------|----------------------------------------------|-----------------------------------------------|
| :white_check_mark: | Open wezterm                                          | Start WSL with zsh in my **default_cwd**  |
| :white_check_mark: | Press Ctrl + T                                            | Open new tab with WSL                             |
| :white_check_mark: | Change to **dev** dir and press Ctrl + T | Open new tab with WSL in **dev** dir       |
| :white_check_mark: | Split pane with keybind                            | New vertical or horizotnal pane with WSL |

**With Poweshell (pwsh)**

| Status                      | Action                                                                                      | Want                                                              | Happens                                                       |
|----------------------|-------------------------------------------------------------------|-------------------------------------------------|------------------------------------------------| 
| :white_check_mark: | From launcher open pwsh                                                       | Start pwsh in the cwd specified in launcher | That I need                                                    |
| :x:                            | From pwsh window I press Ctrl + T                                          | Open new tab with pwsh                             | Open new tab with cmd.exe                         |
| :x:                            | From pwsh window change to **test** dir and press Ctrl + T | Open new tab with pwsh in **test** dir       | Open new tab with cmd.exe in **test** dir   |
| :x:                            | Split pane with keybind                                                            | New vertical or horizotnal pane with pwsh | New pane with cmd.exe                                |

In short, I think I need to specify something like **_"powershell_domains"_** similar to what is done for **_wsl_domains_** or _**ssh_domains**_, because I don't have my Powershell path declared anywhere (apart from the launcher menu table, but it's only used when I open a new instance from the launcher menu, it is not used for new tab or split pane)

<details>
<summary>This is a minimal config for replicate this:</summary>

```lua
return {
   default_domain = "WSL:Ubuntu-20.04",
   wsl_domains = {
      {
         name = "WSL:Ubuntu-20.04",
         distribution = "Ubuntu-20.04",
         username = "pw",
         default_cwd = "/home/pw/Dev",
         default_prog = { "zsh" }
      }
   },
   launch_menu  = {
      {
         label = "wsl",
         args = { "wsl.exe" },
         domain = { DomainName = "local" },
         cwd = "\\\\wsl$\\Ubuntu-20.04\\home\\pw\\Dev"
      },
      {
         label = "pwsh",
         args = { os.getenv("LOCALAPPDATA") .. "\\Microsoft\\WindowsApps\\pwsh.exe", "-nologo" },
         domain = { DomainName = "local" },
         cwd = "D:\\Dev",
      }
   }
}
```
</details>

Thanks.

**Comment by dhruvinsh** [comment]
Were you able find the reliable solution?

**Reply by pwsandoval** [reply]
No, I stopped using Wezterm for a while and focused on Alacritty for Windows (much more limited of course)
